### PR TITLE
Docs to migrate REST APIs to Twirp

### PR DESCRIPTION
**Pull request by marioizquierdo**
Even though the migration process is a case-by-case scenario, it is still useful to have some basic guidelines for the recommended approach. If the community has useful tools that could be used in some cases, we could add links to those tools in this page too. If there are other guides for migrating from gRPC, Swagger and others, we could add those sections on this document later too.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -54,12 +54,12 @@ safely edit anything inside the [docs](./docs) directory, adding new pages or
...
+Then, to render your changes, run docusaurus's local server. See [Install docusaurus on your machine](https://docusaurus.io/docs/en/installation.html).
 
- 1. [Install docusaurus on your machine](https://docusaurus.io/docs/en/installation.html).
- 2. `cd website`
- 3. `npm start`
- 4. Navigate to http://localhost:3000/twirp.
+1.  `cd website`
+2.  `npm install`
+3.  `npm start`
+4.  Navigate to http://localhost:3000/twirp.
```
Comment (marioizquierdo) [inline CONTRIBUTING.md:62]: This is the only part I updated in this document. The way to install docusaurus changed a little bit and the docs no longer apply. Everything else was prettify in my editor "fixing" the markdown

File: docs/migrate_to_twirp.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+id: migrate_to_twirp
+title: Migrate APIs to Twirp
+sidebar_label: Migrate APIs to Twirp
+---
+
+## Migrate a REST API to Twirp
+
+Migrating existing REST/JSON APIs to Twirp is a case-by-case scenario. Unfortunately, it involves creating a new service, generating a new client, and migrating callers to use the new endpoints in the new client.
+
+Twirp APIs are restricted to requests in the form `POST [<prefix>]/[<package>.]<Service>/<Method>` (see [Routing and Serialization](routing.md)). That limitation is what allows to build simple and consistent API schemas, but it also means that each REST API endpoint will have a different way to be translated into a Twirp endpoint. Most REST/JSON APIs don't have strongly defined schemas, and tend to have quirks and edge cases that are hard to automate.
```
Comment (wmatveyenko) [inline docs/migrate_to_twirp.md:11]: Change "what allows to build" to "what allows building"

File: docs/migrate_to_twirp.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+3.  If the API is too big, it may be better to only migrate a few endpoints first. Twirp APIs are good at evolving over time, you can always add more endpoints later.
+4.  Generate code and implement your new API methods. You may be able to re-use some of the older code from the previous endpoints. One way to do this is to implement the new service in a subfolder inside the same project, and then mount the new Twirp handler in a sub-route (you can use the default `/twirp` prefix, or specify a different prefix path if needed).
+5.  Make sure to track stats about API endpoint usage, so you can see the traffic being migrated to the new endpoints. Using [hooks or interceptors](hooks.md), or HTTP [middleware](mux.md) can be useful.
```
Comment (wmatveyenko) [inline docs/migrate_to_twirp.md:19]: remove comma in "), or HTTP"

File: docs/migrate_to_twirp.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+6.  The API callers (upstream services) need to import the new Twirp client and update their calls to use the new Twirp API. Here is where making the new API similar to the old one will be the most useful. Some services may be able to just replace the old client with the new one and start sending traffic to the new endpoints. If they have a decent staging/canary environment and a good deploy system that allows for quick rollbacks, that could be enough to test that the new API is handling the new traffic properly. In other cases, you will need to implement a slider to migrate traffic slowly (10%, 50%, 100%). The slider can be implemented in different ways, either with if statements, e.g. `if r.Intn(100) < 10 { callNewClient() } else { callOldClient() }`, or in more complex cases, with a custom client that implements the same Twirp service interface, but it is able to split the traffic between the old API and the new API.
```
Comment (wmatveyenko) [inline docs/migrate_to_twirp.md:20]: Change "Here is where making the new API similar to the old one will be the most useful." to "This is why making the new API similar to the old one is useful."

**Review by wmatveyenko — CHANGES_REQUESTED** [review]

**Comment by marioizquierdo** [comment]
@wmatveyenko thanks for the review. I addressed the docs with the suggestions, and took the chance to give it another edit pass.

**Review by wmatveyenko — APPROVED** [review]

**Pull request merged**